roles/haproxy/templates/etc/haproxy/haproxy_openstack.cfg

```diff
@@ -182,3 +182,31 @@ backend {{ name }}
   {% endfor %}
 
 {% endfor %}
+
+# Block for percona. Lifted from
+# https://www.percona.com/doc/percona-xtradb-cluster/5.6/howtos/haproxy.html
+# and
+# https://www.percona.com/doc/percona-xtradb-cluster/5.6/howtos/virt_sandbox.html
+# Mark all but first node as backup, so writes go to a single host.
+#
+# Currently does not protect against a server that is JOINING or DISCONNECTED
+# that will require setting up a http service described in above URLs, but that
+# is beyond the MVP. FIXME
+listen mysql-cluster
+  bind :::3307
+  mode tcp
+  retries 3
+  option tcplog
```

Comment:
The "option tcpka"  may be added, "option tcpka"is helpful for the long session, refer to haproxy offical document.
http://www.haproxy.org/download/1.5/doc/configuration.txt